tests/e2e/k8s/test_operator.go

```diff
@@ -0,0 +1,141 @@
+		require.Nil(s.T(), cl[1].CreateBinding("echo", cl[0]))
+		require.Nil(s.T(), cl[1].CreatePolicy(util.PolicyAllowAll))
+
+		data, err := cl[1].AccessService(httpecho.GetEchoValue, importedService, true, nil)
+		require.Nil(s.T(), err)
+		require.Equal(s.T(), cl[0].Name(), data)
+
+		// Check instance1 deployment succeed
+		instance1 := &clusterlink.Instance{}
+		err = peerResource.Get(context.Background(), "cl-instance"+s.fabric.GetNamespace(), controller.OperatorNamespace, instance1)
+		require.Nil(s.T(), err)
+		require.Equal(s.T(), v1.ConditionTrue, instance1.Status.Controlplane.Conditions[string(clusterlink.DeploymentReady)].Status)
+		require.Equal(s.T(), v1.ConditionTrue, instance1.Status.Dataplane.Conditions[string(clusterlink.DeploymentReady)].Status)
+		require.Equal(s.T(), v1.ConditionTrue, instance1.Status.Ingress.Conditions[string(clusterlink.ServiceReady)].Status)
+
+		// Check instance2 fails
+		cond := instanceCPConditionMatch(peerResource, instance2, clusterlink.DeploymentReady, v1.ConditionFalse)
+		err = wait.For(cond, wait.WithTimeout(time.Second*60))
+		require.Nil(s.T(), err)
+		err = peerResource.Get(context.Background(), "instance2", controller.OperatorNamespace, instance2)
+		require.Nil(s.T(), err)
+		require.Equal(s.T(), v1.ConditionFalse, instance2.Status.Controlplane.Conditions[string(clusterlink.DeploymentReady)].Status)
+		require.Equal(s.T(), v1.ConditionFalse, instance2.Status.Dataplane.Conditions[string(clusterlink.DeploymentReady)].Status)
+
+		// Delete first instance.
+		err = peerResource.Delete(context.Background(), instance1)
+		require.Nil(s.T(), err)
+
+		// Check instance2 succeed.
+		cond = instanceCPConditionMatch(peerResource, instance2, clusterlink.DeploymentReady, v1.ConditionTrue)
+		err = wait.For(cond, wait.WithTimeout(time.Second*60))
+		require.Nil(s.T(), err)
+		err = peerResource.Get(context.Background(), "instance2", controller.OperatorNamespace, instance2)
+		require.Nil(s.T(), err)
+		require.Equal(s.T(), v1.ConditionTrue, instance2.Status.Controlplane.Conditions[string(clusterlink.DeploymentReady)].Status)
+		require.Equal(s.T(), v1.ConditionTrue, instance2.Status.Dataplane.Conditions[string(clusterlink.DeploymentReady)].Status)
+		require.Equal(s.T(), v1.ConditionTrue, instance2.Status.Ingress.Conditions[string(clusterlink.ServiceReady)].Status)
+		_, err = cl[1].AccessService(httpecho.GetEchoValue, importedService, true, nil)
+		require.Equal(s.T(), &services.ConnectionResetError{}, err)
+
+		require.Nil(s.T(), cl[0].CreateExport("echo", &httpEchoService))
+		require.Nil(s.T(), cl[0].CreatePolicy(util.PolicyAllowAll))
+		data, err = cl[1].AccessService(httpecho.GetEchoValue, importedService, true, nil)
+		require.Nil(s.T(), err)
+		require.Equal(s.T(), cl[0].Name(), data)
+	})
+}
+
+// instanceCPConditionMatch is a helper function that can be used to check a specific condition match in the controlplane status.
+func instanceCPConditionMatch(
```

Comment:
Can this function be inlined (perhaps even getting rid of the arguments which are always the same) to the test function?

Reply:
But I called it twice, I prefer not to duplicate the code in this case.

Reply:
That's not what I meant :)
I meant moving the function definition inside the test function. e.g.
`instanceCPConditionMatch := func(...) {}`
Then you can get rid of 3 of its 4 arguments which are fixed.